controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -761,6 +761,12 @@ func (r *BareMetalHostReconciler) registerHost(prov provisioner.Provisioner, inf
 		// No need to create PreprovisioningImage if host is not yet registered
 		// or is externally provisioned
 		preprovImgFormats = nil
+	case metal3v1alpha1.StateDeprovisioning, metal3v1alpha1.StateDeleting:
+		// This is an edge case when the deprovisioning happens because the namespace with BMHs is being deleted.
```

Comment:
This comment is not strictly accurate outside the context of this review. We will get here if the host is deprovisioning or deleting for _any_ reason.
I'd put something like "PreprovisioningImage is not required for deprovisioning when cleaning is disabled".